From 5c8a44dfdae037fbe1d1381b56b02bec59c0a40f Mon Sep 17 00:00:00 2001
From: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
Date: Mon, 3 Aug 2020 15:01:49 +0700
Subject: [PATCH] arm64: dts: hihope-rzg2m: Re-order CMA area and remove lossy
 area

We will disable ECC for CMA area, we should move CMA area to start
address of block size 256MB to not using extra split mode.
Lossy feature can't be used when enable ECC, so that we can remove it.

Signed-off-by: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
---
 .../dts/renesas/r8a774a1-hihope-rzg2m-rev2.dts     | 14 ++++----------
 .../boot/dts/renesas/r8a774a1-hihope-rzg2m.dts     | 14 ++++----------
 .../boot/dts/renesas/r8a774a3-hihope-rzg2m.dts     | 14 ++++----------
 3 files changed, 12 insertions(+), 30 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m-rev2.dts b/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m-rev2.dts
index d1b53bc38566..4163610c56ad 100644
--- a/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m-rev2.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m-rev2.dts
@@ -24,17 +24,11 @@
 		#size-cells = <2>;
 		ranges;
 
-		/* device specific region for Lossy Decompression */
-		lossy_decompress: linux,lossy_decompress@54000000 {
-			no-map;
-			reg = <0x00000000 0x54000000 0x0 0x03000000>;
-		};
-
 		/* global autoconfigured region for contiguous allocations */
-		global_cma: linux,cma@58000000 {
+		global_cma: linux,cma@50000000 {
 			compatible = "shared-dma-pool";
 			reusable;
-			reg = <0x00000000 0x58000000 0x0 0x20000000>;
+			reg = <0x00000000 0x50000000 0x0 0x20000000>;
 			linux,cma-default;
 		};
 
@@ -42,13 +36,13 @@
 		mmp_reserved: linux,multimedia@70000000 {
 			compatible = "shared-dma-pool";
 			reusable;
-			reg = <0x00000000 0x78000000 0x0 0x10000000>;
+			reg = <0x00000000 0x70000000 0x0 0x10000000>;
 		};
 	};
 
 	mmngr {
 		compatible = "renesas,mmngr";
-		memory-region = <&mmp_reserved>, <&lossy_decompress>;
+		memory-region = <&mmp_reserved>;
 	};
 
 	mmngrbuf {
diff --git a/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m.dts b/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m.dts
index 9406a196b89f..c68956629f95 100644
--- a/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774a1-hihope-rzg2m.dts
@@ -24,17 +24,11 @@
 		#size-cells = <2>;
 		ranges;
 
-		/* device specific region for Lossy Decompression */
-		lossy_decompress: linux,lossy_decompress@54000000 {
-			no-map;
-			reg = <0x00000000 0x54000000 0x0 0x03000000>;
-		};
-
 		/* global autoconfigured region for contiguous allocations */
-		global_cma: linux,cma@58000000 {
+		global_cma: linux,cma@50000000 {
 			compatible = "shared-dma-pool";
 			reusable;
-			reg = <0x00000000 0x58000000 0x0 0x20000000>;
+			reg = <0x00000000 0x50000000 0x0 0x20000000>;
 			linux,cma-default;
 		};
 
@@ -42,13 +36,13 @@
 		mmp_reserved: linux,multimedia@70000000 {
 			compatible = "shared-dma-pool";
 			reusable;
-			reg = <0x00000000 0x78000000 0x0 0x10000000>;
+			reg = <0x00000000 0x70000000 0x0 0x10000000>;
 		};
 	};
 
 	mmngr {
 		compatible = "renesas,mmngr";
-		memory-region = <&mmp_reserved>, <&lossy_decompress>;
+		memory-region = <&mmp_reserved>;
 	};
 
 	mmngrbuf {
diff --git a/arch/arm64/boot/dts/renesas/r8a774a3-hihope-rzg2m.dts b/arch/arm64/boot/dts/renesas/r8a774a3-hihope-rzg2m.dts
index e6bf48d43e98..ef2ce1f85d30 100644
--- a/arch/arm64/boot/dts/renesas/r8a774a3-hihope-rzg2m.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774a3-hihope-rzg2m.dts
@@ -24,17 +24,11 @@
 		#size-cells = <2>;
 		ranges;
 
-		/* device specific region for Lossy Decompression */
-		lossy_decompress: linux,lossy_decompress@54000000 {
-			no-map;
-			reg = <0x00000000 0x54000000 0x0 0x03000000>;
-		};
-
 		/* global autoconfigured region for contiguous allocations */
-		global_cma: linux,cma@58000000 {
+		global_cma: linux,cma@50000000 {
 			compatible = "shared-dma-pool";
 			reusable;
-			reg = <0x00000000 0x58000000 0x0 0x20000000>;
+			reg = <0x00000000 0x50000000 0x0 0x20000000>;
 			linux,cma-default;
 		};
 
@@ -42,13 +36,13 @@
 		mmp_reserved: linux,multimedia@70000000 {
 			compatible = "shared-dma-pool";
 			reusable;
-			reg = <0x00000000 0x78000000 0x0 0x10000000>;
+			reg = <0x00000000 0x70000000 0x0 0x10000000>;
 		};
 	};
 
 	mmngr {
 		compatible = "renesas,mmngr";
-		memory-region = <&mmp_reserved>, <&lossy_decompress>;
+		memory-region = <&mmp_reserved>;
 	};
 
 	mmngrbuf {
-- 
2.17.1

